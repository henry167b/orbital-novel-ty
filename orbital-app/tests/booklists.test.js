import React from "react";
import renderer from "react-test-renderer";

import BookLists from "../app/(home)/booklists";

jest.mock("expo-router", () => ({
  useRouter: () => ({
    replace: jest.fn(),
  }),
}));

describe("<BookLists />", () => {
  it("can render", () => {
    const tree = renderer.create(<BookLists />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});