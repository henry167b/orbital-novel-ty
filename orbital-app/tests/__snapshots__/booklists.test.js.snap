// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookLists /> can render 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "paddingHorizontal": 35,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "white",
          "elevation": 0,
          "paddingHorizontal": 0,
          "paddingTop": 60,
          "zIndex": 0,
        },
        {},
        {},
      ]
    }
    testID="appbar-header-root-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "white",
          "height": 64,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": "100%",
        }
      }
      testID="appbar-header-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "paddingBottom": undefined,
            "paddingHorizontal": 4,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="appbar-header"
      >
        <View
          style={
            {
              "width": 52,
            }
          }
        />
        <View
          pointerEvents="box-none"
          style={
            [
              {
                "flex": 1,
                "paddingHorizontal": 12,
              },
              {
                "paddingHorizontal": 0,
              },
              [
                false,
                {
                  "alignItems": "center",
                },
                undefined,
              ],
            ]
          }
          testID="appbar-content"
        >
          <Text
            accessibilityRole="header"
            accessibilityTraits="header"
            accessible={true}
            numberOfLines={1}
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontSize": 22,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 28,
                  },
                  false,
                  undefined,
                ],
              ]
            }
            testID="appbar-content-title-text"
          >
            Book Lists
          </Text>
        </View>
        <View
          style={
            {
              "width": 52,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderBottomRightRadius": 0,
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 20,
            "borderRightWidth": 0,
            "borderTopRightRadius": 0,
            "borderWidth": 1,
          },
          {
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 76,
          },
          [
            {
              "borderRadius": 20,
            },
            {},
          ],
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderBottomRightRadius": 0,
              "borderRadius": 20,
              "borderRightWidth": 0,
              "borderTopRightRadius": 0,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 16,
                "paddingVertical": 9,
              },
              {
                "paddingVertical": 9,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "textAlign": "center",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                ],
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  {
                    "fontSize": 10,
                  },
                ]
              }
            >
              WISH LIST
            </Text>
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderBottomLeftRadius": 0,
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 20,
            "borderTopLeftRadius": 0,
            "borderWidth": 1,
          },
          {
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 76,
          },
          [
            {
              "borderRadius": 20,
            },
            {},
          ],
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderBottomLeftRadius": 0,
              "borderRadius": 20,
              "borderTopLeftRadius": 0,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 16,
                "paddingVertical": 9,
              },
              {
                "paddingVertical": 9,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "textAlign": "center",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                ],
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  {
                    "fontSize": 10,
                  },
                ]
              }
            >
              LOANS
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
